src/connector/src/source/base.rs

```diff
@@ -195,6 +195,14 @@ pub fn spawn_data_generation_stream<T: Send + 'static>(
     stream: impl Stream<Item = T> + Send + 'static,
     buffer_size: usize,
 ) -> impl Stream<Item = T> + Send + 'static {
+    static RUNTIME: LazyLock<Runtime> = LazyLock::new(|| {
+        tokio::runtime::Builder::new_multi_thread()
+            .thread_name("risingwave-data-generation")
+            .enable_all()
+            .build()
+            .expect("failed to build data-generation runtime")
+    });
+
```

Comment:
so will all connectors run in a separate runtime apart from executor?

Reply:
Yes. These data generations are usually CPU-intensive and may cause scheduling problems if we put them in the same runtime as streaming actors.